src/javascript/hooks.tsx

```diff
@@ -0,0 +1,117 @@
+// Some useful utility hooks for React 16.8
+import * as React from "react";
+
+// Convert a callback taking a string into a callback taking event.target.value
+export const useEventValue = (setValue: (value: string) => void) =>
+  React.useCallback(
+    (event: React.ChangeEvent<HTMLInputElement>) =>
+      setValue(event.target.value),
+    [setValue]
+  );
+
+// Convert a callback taking a boolean into a callback taking event.target.checked
+export const useEventChecked = (setChecked: (checked: boolean) => void) =>
+  React.useCallback(
+    (event: React.ChangeEvent<HTMLInputElement>) =>
+      setChecked(event.target.checked),
+    [setChecked]
+  );
+
+// Same as `useState`, but also store the value in `localStorage` with a
+// `key` whenever it is updated, and initialize the state from `localStorage`,
+// if present. If `subscribe` is true, also propogate changes from localStorage
+// from other tabs into the state.
+export const useLocalStorage = function(
```

Comment:
If the values you store here will ever change, you probably want to add in an optional 'migration' functionality of some type. Even if it's just a callback that can be used to invalidate old data. Adding the 'version' that the data was stored with as a part of the key or something probably wouldn't be a bad idea either.